lib/bamboo/adapters/send_grid_adapter.ex

```diff
@@ -440,4 +441,12 @@ defmodule Bamboo.SendGridAdapter do
     do: Map.put(body, :ip_pool_name, ip_pool_name)
 
   defp put_ip_pool_name(body, _), do: body
+
+  defp put_unique_args(body, %Email{private: %{unique_args: unique_args}})
```

Comment:
Thanks for adding this function. Could you add a test to ensure this behaves as expected? I know we tested the `with_unique_args` function in the helper, but it would be nice to test this one too.

Reply:
@jeepers3327 whenever you have time, if we add a test for this, I think the PR is good to go. 👍

Reply:
@germsvel I've added a test case for `send_grid_adapter`. Let me know if I need to a add more tests.